src/test/java/com/lafaspot/pop/client/PopClientProdIT.java

```diff
@@ -47,6 +55,74 @@ public void init() throws PopException {
     }
 
     @Test
+    public void testChannelPipeline() throws PopException, InterruptedException, ExecutionException {
+        System.out.println("testChannelPipeline Begin");
+        PopSession sess = client.createSession();
+        PopFuture<PopCommandResponse> f = sess.connect(server, port, 120000, 120000);
+        f.get();
+        final Channel ch = sess.getChannel();
+        Assert.assertNotNull(ch);
+        final ChannelPipeline pipe = ch.pipeline();
+        Assert.assertNotNull(pipe);
+        final String[] handlerList = { PopSession.SSL_HANDLER, PopSession.INACTIVITY_HANDLER, PopSession.DELIMITER, PopSession.DECODER,
+                PopSession.ENCODER, PopSession.POP_HANDLER };
+        final Iterator<Entry<String, ChannelHandler>> it = pipe.iterator();
+        int index = 0;
+        while (it.hasNext()) {
+            final Entry<String, ChannelHandler> entry = it.next();
+            Assert.assertNotNull(entry);
+            final String name = entry.getKey();
+            Assert.assertEquals(name, handlerList[index++]);
+            Assert.assertNotNull(entry.getValue());
+        }
+        System.out.println("testChannelPipeline DONE");
```

Comment:
complete the test properly by doing a disconnect, validating that disconnect worked properly

Reply:
Added the disconnect verification.